### ISSUE TITLE
Signal definition with signalExpression fails to parse when root-level signals are also defined

### ISSUE THREAD

**Issue by hun10** [flowable/flowable-engine#1827]
When process definition contains both root-level and signal-expression signal definitions, it fails to parse:

```
org.flowable.common.engine.api.FlowableException: Error parsing XML

...

Caused by: java.lang.NullPointerException
	at org.flowable.bpmn.model.BpmnModel.getSignal(BpmnModel.java:351)
	at org.flowable.engine.impl.bpmn.parser.handler.IntermediateThrowEventParseHandler.executeParse(IntermediateThrowEventParseHandler.java:48)
	at org.flowable.engine.impl.bpmn.parser.handler.IntermediateThrowEventParseHandler.executeParse(IntermediateThrowEventParseHandler.java:28)
	at org.flowable.engine.impl.bpmn.parser.handler.AbstractBpmnParseHandler.parse(AbstractBpmnParseHandler.java:57)
	at org.flowable.engine.impl.bpmn.parser.handler.AbstractActivityBpmnParseHandler.parse(AbstractActivityBpmnParseHandler.java:33)
	at org.flowable.engine.impl.bpmn.parser.BpmnParseHandlers.parseElement(BpmnParseHandlers.java:79)
	at org.flowable.engine.impl.bpmn.parser.BpmnParse.processFlowElements(BpmnParse.java:318)
	at org.flowable.engine.impl.bpmn.parser.handler.ProcessParseHandler.transformProcess(ProcessParseHandler.java:76)
	at org.flowable.engine.impl.bpmn.parser.handler.ProcessParseHandler.executeParse(ProcessParseHandler.java:49)
	at org.flowable.engine.impl.bpmn.parser.handler.ProcessParseHandler.executeParse(ProcessParseHandler.java:33)
	at org.flowable.engine.impl.bpmn.parser.handler.AbstractBpmnParseHandler.parse(AbstractBpmnParseHandler.java:57)
	at org.flowable.engine.impl.bpmn.parser.BpmnParseHandlers.parseElement(BpmnParseHandlers.java:79)
	at org.flowable.engine.impl.bpmn.parser.BpmnParse.applyParseHandlers(BpmnParse.java:282)
	at org.flowable.engine.impl.bpmn.parser.BpmnParse.execute(BpmnParse.java:195)
	... 57 more
```

Sample process definition for the test case:

```
<?xml version="1.0" encoding="UTF-8"?>
<definitions id="definitions"
	xmlns="http://www.omg.org/spec/BPMN/20100524/MODEL"
	xmlns:activiti="http://activiti.org/bpmn" 
	targetNamespace="Examples">

	<signal id="someSignal" name="someSignalName"/>

	<process id="throwSignal">

		<startEvent id="start" />

		<sequenceFlow sourceRef="start" targetRef="signalEvent" />

		<intermediateThrowEvent id="signalEvent" name="Alert" >
			<signalEventDefinition activiti:signalExpression="${mySignalName}" />			
		</intermediateThrowEvent>

		<sequenceFlow sourceRef="signalEvent" targetRef="end" />

		<endEvent id="end" />

	</process>

</definitions>
```

**Comment by tijsrademakers** [comment]
Merged the PR, thanks

**Comment by hun10** [comment]
@tijsrademakers Will this be available in any of 6.4.x releases?

**Comment by tijsrademakers** [comment]
@hun10 we are thinking of doing a 6.4.2 release and then it will be included indeed.
